### DISCUSSION TITLE
New to rust, but no syntax highlighting after installing rust-analyzer

### DISCUSSION THREAD

**Discussion by johnend** [rust-lang/rust-analyzer#17396] [Q&A] [answered]
I was looking to learn rust and got rustlings installed and ready to go. Opened the files within NeoVim after installing some basics and get my syntax highlighting.

However, after installing rust-analyzer syntax highlighting is gone - literally no colors just comment highlights (everything grey). I assumed this was the semantic tokes stuff that is talked about in various places so I added this to my config:
```lua
for _, group in ipairs(vim.fn.getcompletion("@lsp", "highlight")) do
      vim.api.nvim_set_hl(0, group, {})
    end
```
But this did nothing.

Then I noticed that if I run the `RustFmt` command the colors came back (although not permanently). Interesting.

Then after trying to get it working on opening a file I also noticed that it does show highlighting when initially opening the file, and then while the LSP is loading it goes away. See below gif:

![ScreenRecording2024-06-12at00 04 42-ezgif com-video-to-gif-converter](https://github.com/rust-lang/rust-analyzer/assets/3886057/d0c1f6db-c9e9-4295-b4d7-be2ae7555da7)

Interestingly enough then when I go into insert mode, and then make a small change (insert a line, add a space - something small) it comes back again. Strange.

But then going back to normal mode it goes away again.

Would really appreciate some help figuring this one out 😂

**Comment by lnicola** [comment, marked as answer]
Did you run `rustlings lsp`?

**Reply by johnend** [reply]
That was it 🤦  need to RTM more thoroughly 😂
